tools/benchmark/src/Configuration.ts

```diff
@@ -184,14 +184,15 @@ export type HookFunction = () => void | Promise<unknown>;
 
 /**
  * Arguments that can be passed to `benchmark` for optional test setup/teardown. Hooks execute once per test body
- * (*not* on each cycle or sample). Hooks--along with the benchmarked function--are run without additional error
+ * (*not* on each cycle or iteration). Hooks--along with the benchmarked function--are run without additional error
  * validation. This means any exception thrown from either a hook or the benchmarked function will cause test
  * failure, and subsequent operations won't be run.
  * @public
  */
 export interface HookArguments {
 	before?: HookFunction;
 	after?: HookFunction;
+	onCycle?: HookFunction;
```

Comment:
might be good to link to https://benchmarkjs.com/docs/#options_onCycle here

Reply:
Yeah, and I'd say add our own description of what the cycle is. I found the official BenchmarkJS docs (or what I could find that wasn't StackOverflow, at least) basically non-existent, regarding the lifecycle of a test and what exactly the cycle is or what it encompasses.